internal/artifact/artifact.go

```diff
@@ -67,13 +67,20 @@ func (t Type) String() string {
 
 // Artifact represents an artifact and its relevant info
 type Artifact struct {
-	Name   string
-	Path   string
-	Goos   string
-	Goarch string
-	Goarm  string
-	Type   Type
-	Extra  map[string]interface{}
+	Name       string
+	Path       string
+	Goos       string
+	Goarch     string
+	Goarm      string
+	Type       Type
+	UploadHash string
```

Comment:
can we move this into `Extra`?

Reply:
sure, I can do that :) Makes sense because it's a gitlab only thing. Maybe there will be more in the future for other vcs.

Reply:
Yeap... its not documented anywhere, but my idea was to use extra for things that not common between all things...